### ISSUE TITLE
csv-parse changes passed options argument

### ISSUE THREAD

**Issue by szebrowski** [wdavidw/node-csv-parse#53]
Hi,
I think that changing passed options object argument inside of lib is bad practice. Because of this I had bug that only manifested itself after second run of function that used parse (I got my options object as module's global variable). 

Also I think that making default immutable options object is common scenario and doing something like this fails:
Example:

``` javascript
var options = {
    delimiter: ';',
    columns  : true
};

var opt = Object.freeze(options);

module.exports = function(options){
   if(!options) options = opt;
   parse(options, ...);
}

// Unhandled rejection TypeError: Can't add property objectMode, object is not extensible
```

**Comment by wdavidw** [comment]
I'm currently [cloning the options](https://github.com/wdavidw/node-csv-parse/blob/master/src/index.coffee.md#parseroptions) and it surprises me you got this bug. Looking at the doc, you dont need to get the returned object as it is the same as the object pass as an argument to freeze. Also,  because the csv-parse module will later modify the options object, we'll get a "TypeError" if the options object is frozen. In any case, it seems that you've got a problem that shall be fixed. Would you be kind enough to submit a patch with a test illustrating the bug and a proposed fix.

**Comment by wdavidw** [comment]
no test illustrating the issue, closing. thank you for reporting

**Comment by elyobo** [comment]
Before you clone the options you modify them by setting `objectMode` to true, so if that was the change that @szebrowski was seeing then it makes sense.

```
Parser = (options = {}) ->
  options.objectMode = true
  @options = {}
  for k, v of options
    @options[k] = v
```

**Comment by wdavidw** [comment]
Fixed in f87c2735907c98dbc39aa65a8fd1e07916f3e6c2
